driver/src/main/scala/api/commands/bson/rwcommands.scala

```diff
@@ -79,7 +79,7 @@ object BSONCommonWriteCommandsImplicits {
       extends DealingWithGenericCommandErrorsReader[DefaultWriteResult] {
     def readResult(doc: BSONDocument): DefaultWriteResult = {
       DefaultWriteResult(
-        ok = doc.getAs[Int]("ok").exists(_ != 0),
+        ok = doc.getAs[BSONBooleanLike]("ok").fold(true)(_.toBoolean),
```

Comment:
In MongoDB 3.4, the `ok` property with boolean value `0|1` from the result has been changed from an `int` to a `double` (why?), so need to use the conversion provided by `BSONBooleanLike` as a compatibility workaround.